src/comm.c

```diff
@@ -62,8 +63,13 @@ static void request_update() {
 static void in_received_handler(DictionaryIterator *received, void *context) {
   phone_contact = true;
   update_in_progress = false;
-  schedule_update(UPDATE_FREQUENCY);
-
+  Tuple* recency = dict_find(received, APP_KEY_RECENCY);
+  if (recency) {
+    int32_t next_update = UPDATE_FREQUENCY - recency->value->uint32 * 1000;
+    int32_t delay = next_update < 60000 ? 60000 : next_update;
```

Comment:
nit: can you move this constant into `config.h` and contrast it with `UPDATE_FREQUENCY`? I don't know, `DEFAULT_UPDATE_FREQUENCY` vs. `AFTER_SGV_DATA_UPDATE_FREQUENCY`?